tactics/tacticals.ml

```diff
@@ -434,9 +434,13 @@ module New = struct
     tclIFTHENELSE t1 (tclTRY t2) t2
 
   (* Try the first tactic that does not fail in a list of tactics *)
-  let rec tclFIRST = function
-    | [] -> tclZEROMSG (str"No applicable tactic.")
-    |  t::rest -> tclORELSE0 t (tclFIRST rest)
+  let rec tclFIRST ?(msg = str "No applicable tactic.") = function
+    | [] -> tclZEROMSG msg
+    | t::rest -> tclORELSE0 t (tclFIRST ~msg rest)
+
+  let rec tclMULTIFIRST ?(msg = str "No applicable tactic.") = function
```

Comment:
Yes. Delay: it's better for recursive definition as it computes the list of tactics to apply lazily. It also allows infinite disjunctions. In the case of `tclANY` it's probably irrelevant (well, if yo have an list of size _n_, yo will apply your first tactic in O(1) step, instead of O(n) with `tclOR`, so in optimistic cases it may be better).

At any rate @ppedrot wrote `tclANY` (I'm the one who introduced `tclORD` in it, though). So I can say I like the name :) .